insights/parsers/krb5.py

```diff
@@ -0,0 +1,108 @@
+"""
+krb5 Configuration file - File
+==============================
+krb5 Configuration are ``/etc/krb5.conf`` and ``/etc/krb5.conf.d/*``,
+and its content format is ``INI config``.
+
+The content of file ``/etc/krb5.conf`` and ``/etc/krb5.conf.d/*`` is
+recorded via INI format.
+
+Typical output of the ``/etc/krb5.conf`` is::
+
+Example:
+    >>> krb5_content = '''
+    [realms]
+     dns_lookup_realm = false
+     ticket_lifetime = 24h
+     default_ccache_name = KEYRING:persistent:%{uid}
+     EXAMPLE.COM = {
+      kdc = kerberos.example.com
+      admin_server = kerberos.example.com
+     }
+     89EXAsdcsdMPLE.COM = {
+      kdc = ks2321serberos.example.com
+      admin_server = kc5433scderberos.example.com
+     }
+     [libdefaults]
+      dnsdsd = false
+      tilnvs = 24h
+      89EXAsd = {
+       kdc = ks2321serberos.exampsdcsm
+       admin_server = kc5433scderberos.exasdscdm
+     }
+    # renew_lifetime = 7d
+    # forwardable = true
+    # rdns = false
+    '''.strip()
+
+    >>> from insights.tests import context_wrap
+    >>> shared = {Krb5Configuration: Krb5Configuration(context_wrap(krb5_content))}
+    >>> krb5_info = shared[Krb5Configuration]
+    >>> krb5_info.data["libdefaults"]["dnsdsd"]
+    "false"
+    >>> krb5_info.data["realms"]["EXAMPLE.COM"]["kdc"]
+    "kerberos.example.com"
+"""
+
+
+from insights.core import Parser
+from insights.core.plugins import parser
+
+
+@parser('krb5')
+@parser('krb5.conf.d')
+class Krb5Configuration(Parser):
```

Comment:
I would recommend, if you're implementing a structure similar to an IniConfigFile parser, that you derive from that and then override the parse_content method as you've done.  That way you get all the methods of IniConfigFile for free.